stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -227,6 +230,207 @@ namespace stan {
       }
     };
 
+    class cholesky_gpu : public vari {
```

Comment:
Please move `cholesky_gpu` to a separate header file. It'll be more in line with the rest of our code base.